go/consensus/api/api.go

```diff
@@ -385,6 +386,24 @@ type Backend interface {
 	//
 	// This may be nil in case checkpoints are disabled.
 	Checkpointer() checkpoint.Checkpointer
+
+	// RegisterP2PBackend registers the P2P backend used for light client state sync.
+	RegisterP2PBackend(rpc.P2P) error
+
+	// LightClient returns the libp2p backend light client (if configured.).
+	LightClient() (LightClient, error)
```

Comment:
I noticed that we store the light client instance in the `Node` and then pass it around. Does it make more sense for the light client to be decoupled from the consensus backend (I guess one reason the consensus backend needs it is for state sync)? Or in case it is coupled should all the places just always access it via `.Consensus.LightClient()`?

Reply:
> Or in case it is coupled should all the places just always access it via .Consensus.LightClient()?

Yeah, if it stays coupled, this should be done.

> Does it make more sense for the light client to be decoupled from the consensus backend

Not sure, maybe. I'll try it out and see how it looks.

Reply:
I decoupled the light-client from the consensus backend, please take another look.